### PR TITLE
Feature/images screen unit tests

### PR DESCRIPTION
**Pull request by michal1990**
Missing unit tests.

### CODE REVIEW

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21)
+@SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION")
+public class EditActivityTest {
+
+    private EditActivity activity;
+
+    @Test
+    public void whenActivityIsStartedWithIntentAndIntentContainsData_thenDataAreFetchedCorrectly() {
```
Comment (BartSoftware) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java]: "_then" is redundant
"DataAre" -> "DataIs"

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        //given
+        ImageDetails imageDetails = new ImageDetails("Path", "Name");
+        Intent editActivityIntent = EditActivity.buildIntent(RuntimeEnvironment.application, imageDetails);
```
Comment (BartSoftware) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java]: RuntimeEnvironment.application could be extracted as a class member (context) and initiated on @Before

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/images/ImagesProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        when(contentResolver.query(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, columns, null, null, null)).thenReturn(cursor);
+        when(cursor.moveToFirst()).thenReturn(true);
+        when(cursor.getString(0)).thenReturn("Title");
+        when(cursor.getString(1)).thenReturn("Data");
+        List<ImageDetails> imagesList = imagesProvider.getImagesList();
+
+        //then
+        assertEquals(1, imagesList.size());
+        assertNotNull(imagesList.get(0));
+        assertEquals("Title", imagesList.get(0).getName());
```
Comment (BartSoftware) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/images/ImagesProviderTest.java]: better approach is to expose this to variable like:
String imageName = imagesList.get(0).getName();

Its my easier to read this assertion:
assertEquals(imageName, "Title");

And imho it is better to set expected value as second parameter, but its not a blocker anyway

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesItemAdapterTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    @Test
+    public void whenViewHolderIsCreatedWithNotEmptyList_thenAdapterWillHaveResults() {
+        //given
+        ViewGroup parent = mock(ViewGroup.class);
+        when(parent.getContext()).thenReturn(activity);
+        ImagesItemAdapter.ViewHolder viewHolder = imagesItemAdapter.onCreateViewHolder(parent, 0);
+
+        ImageDetails imageDetails = new ImageDetails("Path", "Name");
+        List<ImageDetails> imagesList = new ArrayList<>();
+        imagesList.add(imageDetails);
```
Comment (BartSoftware) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesItemAdapterTest.java]: That list could be exposed as member of this class like:
private static final List< ImageDetails > IMAGES_DETAILS_LIST = ImmutableList.of(new ImageDetails("path", "name"));

**Review by BartSoftware — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java [hunk 5]
```diff
@@ -12,17 +12,30 @@
...
         setContentView(R.layout.activity_images);
 
         ImagesCompoundRecyclerView imagesCompoundRecyclerView =
                 (ImagesCompoundRecyclerView) findViewById(R.id.images_compound_recycler_view);
-        new ImagesViewModel(
-                imagesCompoundRecyclerView,
-                new ImagesProvider(getContentResolver()),
-                this);
+
+        Object lastNonConfigurationInstanceObject = getLastCustomNonConfigurationInstance();
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java]: I think it is better to not use configuration retaining for now.

**Pull request merged**